<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Ranges associated to nucleosomes detected by the NOrMAL software using
syntetic reads generated using a normal distribution.
For demonstration purpose. — NOrMAL_nucleosome_ranges • consensusSeekeR</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Ranges associated to nucleosomes detected by the NOrMAL software using
syntetic reads generated using a normal distribution.
For demonstration purpose. — NOrMAL_nucleosome_ranges"><meta property="og:description" content="Ranges associated to nucleosomes detected by the NOrMAL software using
syntetic reads generated using a normal distribution with a variance
of 20 for regions chr1:10000-15000."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">consensusSeekeR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.15.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../articles/consensusSeekeR.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul><ul class="nav navbar-nav navbar-right"><li>
  <a href="https://github.com/adeschen/consensusSeekeR/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Ranges associated to nucleosomes detected by the NOrMAL software using
syntetic reads generated using a normal distribution.
For demonstration purpose.</h1>
    <small class="dont-index">Source: <a href="https://github.com/adeschen/consensusSeekeR/blob/HEAD/R/consensusSeekeR.R" class="external-link"><code>R/consensusSeekeR.R</code></a></small>
    <div class="hidden name"><code>NOrMAL_nucleosome_ranges.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>Ranges associated to nucleosomes detected by the NOrMAL software using
syntetic reads generated using a normal distribution with a variance
of 20 for regions chr1:10000-15000.</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">NOrMAL_nucleosome_ranges</span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="format">
    <h2>Format</h2>
    <p>A <code>GRanges</code> containing one entry per detected
nucleosome. The ranges are surronding the nucleosomes present in the dataset
<code>NOrMAL_nucleosome_positions</code>. The genomic ranges have been obtained by
adding 73 bps on each side of the detected positions.</p>
    </div>
    <div id="references">
    <h2>References</h2>
    
<ul><li><p>Polishko A, Ponts N, Le Roch KG and Lonardi S. 2012. NOrMAL:
Accurate nucleosome positioning using a modified Gaussian mixture
model. Bioinformatics 28 (12): 242-49.</p></li>
</ul></div>
    <div id="see-also">
    <h2>See also</h2>
    <div class="dont-index">
<ul><li><p><code><a href="NOrMAL_nucleosome_positions.html">NOrMAL_nucleosome_positions</a></code>  the associate
genomic positions dataset.</p></li>
<li><p><code><a href="findConsensusPeakRegions.html">findConsensusPeakRegions</a></code> for extracting regions
sharing nucleosomes from more than one experiment.</p></li>
</ul></div>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Loading datasets</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">PING_nucleosome_positions</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">PING_nucleosome_ranges</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">NOrMAL_nucleosome_positions</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">NOrMAL_nucleosome_ranges</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">NucPosSimulator_nucleosome_positions</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">NucPosSimulator_nucleosome_ranges</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Assigning experiment name to each row of the dataset.</span></span></span>
<span class="r-in"><span><span class="co">## Position and range datasets from the same sofware must</span></span></span>
<span class="r-in"><span><span class="co">## have identical names.</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">PING_nucleosome_positions</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"PING"</span>,</span></span>
<span class="r-in"><span>                            <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">PING_nucleosome_positions</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">PING_nucleosome_ranges</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"PING"</span>,</span></span>
<span class="r-in"><span>                            <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">PING_nucleosome_ranges</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">NOrMAL_nucleosome_positions</span><span class="op">)</span> <span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"NOrMAL"</span>,</span></span>
<span class="r-in"><span>                            <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">NOrMAL_nucleosome_positions</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">NOrMAL_nucleosome_ranges</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"NOrMAL"</span>,</span></span>
<span class="r-in"><span>                            <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">NOrMAL_nucleosome_ranges</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">NucPosSimulator_nucleosome_positions</span><span class="op">)</span> <span class="op">&lt;-</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"NucPosSimulator"</span>,</span></span>
<span class="r-in"><span>                            <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">NucPosSimulator_nucleosome_positions</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">NucPosSimulator_nucleosome_ranges</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"NucPosSimulator"</span>,</span></span>
<span class="r-in"><span>                            <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">NucPosSimulator_nucleosome_ranges</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co">## Calculating consensus regions for chromosome 1</span></span></span>
<span class="r-in"><span><span class="co">## with a default region size of 30 bp (2 * extendingSize).</span></span></span>
<span class="r-in"><span><span class="co">## Consensus regions are resized to include all genomic regions of</span></span></span>
<span class="r-in"><span><span class="co">## included nucleosomes.</span></span></span>
<span class="r-in"><span><span class="co">## Nucleosomes from at least 2 software must be present</span></span></span>
<span class="r-in"><span><span class="co">## in a region to be retained as a consensus region.</span></span></span>
<span class="r-in"><span><span class="va">chrList</span> <span class="op">&lt;-</span> <span class="fu">Seqinfo</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chr1"</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">249250621</span><span class="op">)</span>, <span class="cn">NA</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="findConsensusPeakRegions.html">findConsensusPeakRegions</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>    narrowPeaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">PING_nucleosome_ranges</span>,</span></span>
<span class="r-in"><span>                            <span class="va">NOrMAL_nucleosome_ranges</span>,</span></span>
<span class="r-in"><span>                            <span class="va">NucPosSimulator_nucleosome_ranges</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    peaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">PING_nucleosome_positions</span>,</span></span>
<span class="r-in"><span>                            <span class="va">NOrMAL_nucleosome_positions</span>,</span></span>
<span class="r-in"><span>                            <span class="va">NucPosSimulator_nucleosome_positions</span><span class="op">)</span>,</span></span>
<span class="r-in"><span>    chrInfo <span class="op">=</span> <span class="va">chrList</span>,</span></span>
<span class="r-in"><span>    extendingSize <span class="op">=</span> <span class="fl">15</span>,</span></span>
<span class="r-in"><span>    expandToFitPeakRegion <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>    shrinkToFitPeakRegion <span class="op">=</span> <span class="cn">TRUE</span>,</span></span>
<span class="r-in"><span>    minNbrExp <span class="op">=</span> <span class="fl">2</span>,</span></span>
<span class="r-in"><span>    nbrThreads <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $call</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> findConsensusPeakRegions(narrowPeaks = c(PING_nucleosome_ranges, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     NOrMAL_nucleosome_ranges, NucPosSimulator_nucleosome_ranges), </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     peaks = c(PING_nucleosome_positions, NOrMAL_nucleosome_positions, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>         NucPosSimulator_nucleosome_positions), chrInfo = chrList, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     extendingSize = 15, expandToFitPeakRegion = TRUE, shrinkToFitPeakRegion = TRUE, </span>
<span class="r-out co"><span class="r-pr">#&gt;</span>     minNbrExp = 2, nbrThreads = 1)</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> $consensusRanges</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> GRanges object with 27 ranges and 0 metadata columns:</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>        seqnames      ranges strand</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>           &lt;Rle&gt;   &lt;IRanges&gt;  &lt;Rle&gt;</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [1]     chr1 10002-10148      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [2]     chr1 10167-10314      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [3]     chr1 10334-10482      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [4]     chr1 10503-10649      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    [5]     chr1 10670-10816      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>    ...      ...         ...    ...</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [23]     chr1 14177-14325      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [24]     chr1 14340-14488      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [25]     chr1 14506-14657      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [26]     chr1 14677-14824      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   [27]     chr1 14842-14992      *</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   -------</span>
<span class="r-out co"><span class="r-pr">#&gt;</span>   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> </span>
<span class="r-out co"><span class="r-pr">#&gt;</span> attr(,"class")</span>
<span class="r-out co"><span class="r-pr">#&gt;</span> [1] "consensusRanges"</span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Astrid Deschenes, Pascal Belleau, Arnaud Droit.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer></div>

  


  

  </body></html>

