<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Detection of consensus regions inside a group of experiments • consensusSeekeR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Detection of consensus regions inside a group of experiments">
<meta property="og:description" content="consensusSeekeR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">consensusSeekeR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.31.2</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/consensusSeekeR.html">Get started</a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/adeschen/consensusSeekeR/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Detection of consensus regions inside a group of
experiments</h1>
                        <h4 data-toc-skip class="author">Astrid
Deschênes, Fabien Claude Lamaze, Pascal Belleau and Arnaud Droit</h4>
            
            <h4 data-toc-skip class="date">23 February, 2024</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/adeschen/consensusSeekeR/blob/HEAD/vignettes/consensusSeekeR.Rmd" class="external-link"><code>vignettes/consensusSeekeR.Rmd</code></a></small>
      <div class="hidden name"><code>consensusSeekeR.Rmd</code></div>

    </div>

    
        <div class="abstract">
      <p class="abstract">Abstract</p>
      This package compares genomic positions and genomic ranges from
      multiple experiments to extract common regions. The size of the
      analyzed region is adjustable as well as the number of experiences
      in which a feature must be present in a potential region to tag
      this region as a consensus region. In genomic analysis where
      feature identification generates a position value surrounded by a
      genomic range, such as ChIP-Seq peaks and nucleosome positions,
      the replication of an experiment may result in slight differences
      between predicted values. This package enables the conciliation of
      the results into consensus regions.
    </div>
    
<script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  document.querySelector("h1").className = "title";
});
</script><script type="text/javascript">
document.addEventListener("DOMContentLoaded", function() {
  var links = document.links;  
  for (var i = 0, linksLength = links.length; i < linksLength; i++)
    if (links[i].hostname != window.location.hostname)
      links[i].target = '_blank';
});
</script><p><br></p>
<div class="section level2">
<h2 id="licensing-and-citing">Licensing and citing<a class="anchor" aria-label="anchor" href="#licensing-and-citing"></a>
</h2>
<p>This package and the underlying <em><a href="https://bioconductor.org/packages/3.15/consensusSeekeR" class="external-link">consensusSeekeR</a></em>
code are distributed under the Artistic license 2.0. You are free to use
and redistribute this software.</p>
<p>If you use this package for a publication, we would ask you to cite
the following:</p>
<blockquote>
<p>Samb R, Khadraoui K, Belleau P, et al. (2015) Using informative
Multinomial-Dirichlet prior in a t-mixture with reversible jump
estimation of nucleosome positions for genome-wide profiling.
Statistical Applications in Genetics and Molecular Biology. Published
online before print December 10, 2015.
<a href="http://dx.doi.org/10.1515/sagmb-2014-0098" class="external-link">doi:10.1515/sagmb-2014-0098</a></p>
</blockquote>
</div>
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Genome data, such as genes, nucleosomes or single-nucleotide
polymorphisms (SNPs) are linked to the genome by occupying either a
range of positions or a single position on the sequence. Genomic related
data integration is made possible by treating the data as ranges on the
genome <span class="citation">(Lawrence et al. 2013)</span>.
Bioconductor has developed an infrastructure, including packages such as
<em><a href="https://bioconductor.org/packages/3.15/GenomicRanges" class="external-link">GenomicRanges</a></em>,
<em><a href="https://bioconductor.org/packages/3.15/IRanges" class="external-link">IRanges</a></em>
and <em><a href="https://bioconductor.org/packages/3.15/GenomicFeatures" class="external-link">GenomicFeatures</a></em>,
which facilitate the integrative statistical analysis of range-based
genomic data.</p>
<p>Ranges format is a convenient way for the analysis of diffent
experimental genomic data. As an example, the peak calling step, in the
analysis of ChIP-seq data, commonly generates NarrowPeak outputs. The
NarrowPeak format, wich is used by the ENCODE project <span class="citation">(Dunham et al. 2012)</span>, includes a peak position
located inside a genomic range.</p>
<p>In genomic analysis where feature identification generate a position
value surrounded by a genomic range, such as ChIP-Seq peaks and
nucleosome postions, the replication of an experiment may result in
slight differences between predicted values. Conciliation of the results
can be difficult, especially when many replicates are done. One current
approach used to identify consensus regions in a group of results
consist of extracting the overlapping regions of the genomic ranges.
This approach, when used on a large number of experiments, can miss, as
a side effect, regions when one of the experiment has missing or
slightly shift features. On the other hand, the use of the union of the
regions can result in wide consensus ranges.</p>
<p>As an example, the Figure @ref(fig:peakRegion) shows, using
Integrative genomics viewer <span class="citation">(Robinson et al.
2011)</span>, two ChIP-Seq peaks from <a href="https://www.encodeproject.org" class="external-link">ENCODE</a> for the FOSL2
transcription factor (DCC accession: ENCFF002CFN). The data have been
analyzed using MACS2 <span class="citation">(Zhang et al. 2008)</span>
with the default parameters and the q-value set to 0.05. The ChIP-Seq
peak is a genomic feature that can be defined by a position value (the
peak position) and a genome range (the enriched region). This example
shows that the peak position is not necessarily at the center of the
enriched region.</p>
<div class="figure">
<img src="figures/peak_and_region.jpg" alt=""><p class="caption">Two ChIP-Seq peaks. A ChIP-seq peak is defined by a
enriched region an a peak position.</p>
</div>
<p>The <em><a href="https://bioconductor.org/packages/3.15/consensusSeekeR" class="external-link">consensusSeekeR</a></em>
package implements a novative way to identify consensus which use the
features positions, instead of the most commonly used genomic
ranges.</p>
</div>
<div class="section level2">
<h2 id="the-consensusseeker-package">The consensusSeekeR package<a class="anchor" aria-label="anchor" href="#the-consensusseeker-package"></a>
</h2>
<p>The <em><a href="https://bioconductor.org/packages/3.15/consensusSeekeR" class="external-link">consensusSeekeR</a></em>
package implements a novative way to identify consensus ranges in a
group of experiments which generated position values surrounded by
genomic ranges. The <em><a href="https://bioconductor.org/packages/3.15/consensusSeekeR" class="external-link">consensusSeekeR</a></em>
package is characterized by its use of the position values, instead of
the genomic ranges, to identify the consensus regions. The positions
values have the double advantages of being, most of the time, the most
important information from features and allowing creation of consensius
regions of smaller ranges.</p>
<p>Using iterative steps on ordered features position values from all
experiments, a window of fixed size (specified by user) with the current
feature position as starting point is set. All features which reside
inside the window are gathered to calculate a median feature position
which is then used to recreate a new window. This time, the new window
has twice the size fixed by user and its center is the median feature
position. An update of the features located inside the window is done
and the median feature position is recalculated. This step is repeated
up to the moment that the set of features remains identical between two
iterations. The final set of features positions is used to fix the
central position of the consensus region. This final region must respect
the minimum number of experiments with at least one feature inside it to
be retained as a final consensus region. The minimum number of
experiments is set by the user. At last, the consensus region can be
extended or/and shrinked to fit the regions associated to the position
values present inside. If new features positions are added during the
consensus region resizing, the iterative steps are not reprocessed. It
is possible that the extension step adds new features in the extended
consensus region. However, those new features ranges won’t be taken into
account during the extension step.</p>
</div>
<div class="section level2">
<h2 id="loading-consensusseeker-package">Loading consensusSeekeR package<a class="anchor" aria-label="anchor" href="#loading-consensusseeker-package"></a>
</h2>
<p>As with any R package, the <em><a href="https://bioconductor.org/packages/3.15/consensusSeekeR" class="external-link">consensusSeekeR</a></em>
package should first be loaded with the following command:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/adeschen/consensusSeekeR" class="external-link">consensusSeekeR</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="inputs">Inputs<a class="anchor" aria-label="anchor" href="#inputs"></a>
</h2>
<div class="section level3">
<h3 id="positions-and-ranges">Positions and Ranges<a class="anchor" aria-label="anchor" href="#positions-and-ranges"></a>
</h3>
<p>The main function of the <em><a href="https://bioconductor.org/packages/3.15/consensusSeekeR" class="external-link">consensusSeekeR</a></em>
is <code>findConsensusPeakRegions</code>. The mains inputs of the
<code>findConsensusPeakRegions</code> function are:</p>
<ol style="list-style-type: decimal">
<li>a <code>GRanges</code> of the feature <code>positions</code> of all
experiments with a metadata field called <code>name</code>.</li>
<li>a <code>GRanges</code> of the feature <code>ranges</code> for all
experiments with a metadata field called <code>name</code>.</li>
</ol>
<p>Beware that the <code>GRanges</code> of the feature
<code>ranges</code> is only mandatory if the
<code>expandToFitPeakRegion</code> parameter and/or the
<code>shrinkToFitPeakRegion</code> parameter are set to
<code>TRUE</code>.</p>
<p>Both inputs must satify these conditions:</p>
<ul>
<li>All rows of each <code>GRanges</code> must be named after the
experiment source. All entries from the same experiment must be assigned
the same name.</li>
<li>Each feature must have one entry in both <code>GRanges</code>. The
metadata field <code>name</code> is used to associate the feature
position to its range.</li>
</ul>
<p>This is an example showing how a metadata field <code>name</code> can
easily be created and row names can be assigned:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">### Initial dataset without metadata field</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">A549_FOSL2_01_NarrowPeaks_partial</span>, n <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span>
<span><span class="co">## GRanges object with 3 ranges and 0 metadata columns:</span></span>
<span><span class="co">##       seqnames              ranges strand</span></span>
<span><span class="co">##          &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt;</span></span>
<span><span class="co">##   [1]     chr1 249119914-249120249      *</span></span>
<span><span class="co">##   [2]     chr1 249120596-249121174      *</span></span>
<span><span class="co">##   [3]     chr1 249123089-249123453      *</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 25 sequences from an unspecified genome; no seqlengths</span></span>
<span></span>
<span><span class="co">### Adding a new metadata field "name" unique to each entry</span></span>
<span><span class="va">A549_FOSL2_01_NarrowPeaks_partial</span><span class="op">$</span><span class="va">name</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"FOSL2_01_entry_"</span>, </span>
<span>                                <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">A549_FOSL2_01_NarrowPeaks_partial</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">### Assign the same row name to each entry</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">A549_FOSL2_01_NarrowPeaks_partial</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"FOSL2_01"</span>, </span>
<span>                                <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">A549_FOSL2_01_NarrowPeaks_partial</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">### Final dataset with metadata field 'name' and row names</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">A549_FOSL2_01_NarrowPeaks_partial</span>, n <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span>
<span><span class="co">## GRanges object with 3 ranges and 1 metadata column:</span></span>
<span><span class="co">##            seqnames              ranges strand |             name</span></span>
<span><span class="co">##               &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; |      &lt;character&gt;</span></span>
<span><span class="co">##   FOSL2_01     chr1 249119914-249120249      * | FOSL2_01_entry_1</span></span>
<span><span class="co">##   FOSL2_01     chr1 249120596-249121174      * | FOSL2_01_entry_2</span></span>
<span><span class="co">##   FOSL2_01     chr1 249123089-249123453      * | FOSL2_01_entry_3</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 25 sequences from an unspecified genome; no seqlengths</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="chromosomes-information">Chromosomes information<a class="anchor" aria-label="anchor" href="#chromosomes-information"></a>
</h3>
<p>The chromosomes information is mandatory. It ensures that the
consensus regions do not exceed the length of the chromosomes.</p>
<p>The chromosomes information is contained inside a
<code>Seqinfo</code> object. The information from some UCSC genomes can
be fetched automatically using the <em><a href="https://bioconductor.org/packages/3.15/GenomeInfoDb" class="external-link">GenomeInfoDb</a></em>
package.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">### Import library</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/GenomeInfoDb" class="external-link">GenomeInfoDb</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co">### Get the information for Human genome version 19</span></span>
<span><span class="va">hg19Info</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/Seqinfo-class.html" class="external-link">Seqinfo</a></span><span class="op">(</span>genome<span class="op">=</span><span class="st">"hg19"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">### Subset the object to keep only the analyzed chromosomes</span></span>
<span><span class="va">hg19Subset</span> <span class="op">&lt;-</span> <span class="va">hg19Info</span><span class="op">[</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chr1"</span>, <span class="st">"chr10"</span>, <span class="st">"chrX"</span><span class="op">)</span><span class="op">]</span></span></code></pre></div>
<p>A <code>Seqinfo</code> object can also be created using the
chromosomes information specific to the analyzed genome.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">### Create an Seqinfo Object</span></span>
<span><span class="va">chrInfo</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/Seqinfo-class.html" class="external-link">Seqinfo</a></span><span class="op">(</span>seqnames<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"chr1"</span>, <span class="st">"chr2"</span>, <span class="st">"chr3"</span><span class="op">)</span>,</span>
<span>            seqlengths<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1000</span>, <span class="fl">2000</span>, <span class="fl">1500</span><span class="op">)</span>, isCircular<span class="op">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="cn">FALSE</span>, <span class="cn">FALSE</span>, <span class="cn">FALSE</span><span class="op">)</span>,</span>
<span>            genome<span class="op">=</span><span class="st">"BioconductorAlien"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="read-narrowpeak-files">Read NarrowPeak files<a class="anchor" aria-label="anchor" href="#read-narrowpeak-files"></a>
</h2>
<p>The NarrowPeak format is often used to provide called peaks of signal
enrichment based on pooled, normalized data. The <em><a href="https://bioconductor.org/packages/3.15/rtracklayer" class="external-link">rtracklayer</a></em>
package has functions which faciliates the loading of NarrowPeak
files.</p>
<p>Since the main function of the <em><a href="https://bioconductor.org/packages/3.15/consensusSeekeR" class="external-link">consensusSeekeR</a></em>
package needs 2 <code>GRanges</code> objects, some manipulations are
needed to create one <code>GRanges</code> for the regions and one
<code>GRanges</code> for the peaks.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">### Load the needed packages</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va">rtracklayer</span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/GenomicRanges" class="external-link">GenomicRanges</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co">### Demo file contained within the consensusSeekeR package</span></span>
<span><span class="va">file_narrowPeak</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>,</span>
<span>        <span class="st">"A549_FOSL2_ENCSR000BQO_MZW_part_chr_1_and_12.narrowPeak"</span>, package <span class="op">=</span> <span class="st">"consensusSeekeR"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">### Information about the extra columns present in the file need</span></span>
<span><span class="co">### to be specified</span></span>
<span><span class="va">extraCols</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>signalValue <span class="op">=</span> <span class="st">"numeric"</span>, pValue <span class="op">=</span> <span class="st">"numeric"</span>, qValue <span class="op">=</span> <span class="st">"numeric"</span>, peak <span class="op">=</span> <span class="st">"integer"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">### Create genomic ranges for the regions</span></span>
<span><span class="va">regions</span> <span class="op">&lt;-</span> <span class="fu">import</span><span class="op">(</span><span class="va">file_narrowPeak</span>, format <span class="op">=</span> <span class="st">"BED"</span>, extraCols <span class="op">=</span> <span class="va">extraCols</span><span class="op">)</span></span>
<span></span>
<span><span class="co">### Create genomic ranges for the peaks</span></span>
<span><span class="va">peaks</span>           <span class="op">&lt;-</span>  <span class="va">regions</span></span>
<span><span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/range-squeezers.html" class="external-link">ranges</a></span><span class="op">(</span><span class="va">peaks</span><span class="op">)</span>   <span class="op">&lt;-</span>  <span class="fu"><a href="https://rdrr.io/pkg/IRanges/man/IRanges-constructor.html" class="external-link">IRanges</a></span><span class="op">(</span>start <span class="op">=</span> <span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/stats/start.html" class="external-link">start</a></span><span class="op">(</span><span class="va">regions</span><span class="op">)</span> <span class="op">+</span> <span class="va">regions</span><span class="op">$</span><span class="va">peak</span><span class="op">)</span>, width <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">regions</span><span class="op">$</span><span class="va">peak</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">### First rows of each GRanges object</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">regions</span>, n <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="co">## GRanges object with 2 ranges and 6 metadata columns:</span></span>
<span><span class="co">##       seqnames        ranges strand |                   name     score</span></span>
<span><span class="co">##          &lt;Rle&gt;     &lt;IRanges&gt;  &lt;Rle&gt; |            &lt;character&gt; &lt;numeric&gt;</span></span>
<span><span class="co">##   [1]     chr1 846589-846847      * | peaks/Hosa_A549_FOSL..        57</span></span>
<span><span class="co">##   [2]     chr1 856004-856159      * | peaks/Hosa_A549_FOSL..        43</span></span>
<span><span class="co">##       signalValue    pValue    qValue      peak</span></span>
<span><span class="co">##         &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;integer&gt;</span></span>
<span><span class="co">##   [1]     5.59984   8.75159   5.77648        98</span></span>
<span><span class="co">##   [2]     5.16770   7.21902   4.33609       108</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 2 sequences from an unspecified genome; no seqlengths</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">peaks</span>, n <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="co">## GRanges object with 2 ranges and 6 metadata columns:</span></span>
<span><span class="co">##       seqnames    ranges strand |                   name     score signalValue</span></span>
<span><span class="co">##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |            &lt;character&gt; &lt;numeric&gt;   &lt;numeric&gt;</span></span>
<span><span class="co">##   [1]     chr1    846687      * | peaks/Hosa_A549_FOSL..        57     5.59984</span></span>
<span><span class="co">##   [2]     chr1    856112      * | peaks/Hosa_A549_FOSL..        43     5.16770</span></span>
<span><span class="co">##          pValue    qValue      peak</span></span>
<span><span class="co">##       &lt;numeric&gt; &lt;numeric&gt; &lt;integer&gt;</span></span>
<span><span class="co">##   [1]   8.75159   5.77648        98</span></span>
<span><span class="co">##   [2]   7.21902   4.33609       108</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 2 sequences from an unspecified genome; no seqlengths</span></span></code></pre></div>
<p>A simpler way is to use the <code>readNarrowPeakFile</code> function
of the <em><a href="https://bioconductor.org/packages/3.15/consensusSeekeR" class="external-link">consensusSeekeR</a></em>
package which generates both the peaks and the narrowPeak
<code>GRanges</code>.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/adeschen/consensusSeekeR" class="external-link">consensusSeekeR</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co">### Demo file contained within the consensusSeekeR package</span></span>
<span><span class="va">file_narrowPeak</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>,</span>
<span>    <span class="st">"A549_FOSL2_ENCSR000BQO_MZW_part_chr_1_and_12.narrowPeak"</span>, package <span class="op">=</span> <span class="st">"consensusSeekeR"</span><span class="op">)</span></span>
<span></span>
<span><span class="co">### Create genomic ranges for both the regions and the peaks</span></span>
<span><span class="va">result</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/readNarrowPeakFile.html">readNarrowPeakFile</a></span><span class="op">(</span><span class="va">file_narrowPeak</span>, extractRegions <span class="op">=</span> <span class="cn">TRUE</span>, extractPeaks <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co">### First rows of each GRanges object</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">result</span><span class="op">$</span><span class="va">narrowPeak</span>, n <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="co">## GRanges object with 2 ranges and 6 metadata columns:</span></span>
<span><span class="co">##       seqnames        ranges strand |                   name     score</span></span>
<span><span class="co">##          &lt;Rle&gt;     &lt;IRanges&gt;  &lt;Rle&gt; |            &lt;character&gt; &lt;numeric&gt;</span></span>
<span><span class="co">##   [1]     chr1 846589-846847      * | peaks/Hosa_A549_FOSL..        57</span></span>
<span><span class="co">##   [2]     chr1 856004-856159      * | peaks/Hosa_A549_FOSL..        43</span></span>
<span><span class="co">##       signalValue    pValue    qValue      peak</span></span>
<span><span class="co">##         &lt;numeric&gt; &lt;numeric&gt; &lt;numeric&gt; &lt;integer&gt;</span></span>
<span><span class="co">##   [1]     5.59984   8.75159   5.77648        98</span></span>
<span><span class="co">##   [2]     5.16770   7.21902   4.33609       108</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 2 sequences from an unspecified genome; no seqlengths</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">result</span><span class="op">$</span><span class="va">peak</span>, n <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="co">## GRanges object with 2 ranges and 6 metadata columns:</span></span>
<span><span class="co">##       seqnames    ranges strand |                   name     score signalValue</span></span>
<span><span class="co">##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; |            &lt;character&gt; &lt;numeric&gt;   &lt;numeric&gt;</span></span>
<span><span class="co">##   [1]     chr1    846687      * | peaks/Hosa_A549_FOSL..        57     5.59984</span></span>
<span><span class="co">##   [2]     chr1    856112      * | peaks/Hosa_A549_FOSL..        43     5.16770</span></span>
<span><span class="co">##          pValue    qValue      peak</span></span>
<span><span class="co">##       &lt;numeric&gt; &lt;numeric&gt; &lt;integer&gt;</span></span>
<span><span class="co">##   [1]   8.75159   5.77648        98</span></span>
<span><span class="co">##   [2]   7.21902   4.33609       108</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 2 sequences from an unspecified genome; no seqlengths</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="case-study-nucleosome-positioning">Case study: nucleosome positioning<a class="anchor" aria-label="anchor" href="#case-study-nucleosome-positioning"></a>
</h2>
<p>Global gene expression patterns are established and maintained by the
concerted actions of Transcription Factors (TFs) and the proteins that
constitute chromatin. The key structural element of chromatin is the
nucleosome, which consists of an octameric histone core wrapped by 147
bps of DNA and connected to its neighbor by approximately 10-80 pbs of
linker DNA <span class="citation">(Kornberg and Lorch 1999)</span>.
Nucleosome occupancy and positioning have been proved to be dynamic. It
also has a major impact on expression, regulation, and evolution of
eukaryotic genes <span class="citation">(Jiang, Zhang, and Zhang
2015)</span>.</p>
<div class="section level3">
<h3 id="comparing-nucleosome-positioning-results-from-different-software">Comparing nucleosome positioning results from different
software<a class="anchor" aria-label="anchor" href="#comparing-nucleosome-positioning-results-from-different-software"></a>
</h3>
<p>With the development of Next-generation sequencing, nucleosome
positioning using MNase-Seq data or MNase- or sonicated- ChIP-Seq data
combined with either single-end or paired-end sequencing have evolved as
popular techniques. Software such as PING <span class="citation">(Woo et
al. 2013)</span> and NOrMAL <span class="citation">(Polishko et al.
2012)</span>, generates output which contains the positions of the
predicted nucleosomes, which simply are one base pair positions on the
reference genome. This position represents the center of the predicted
nucleosome. A range of <span class="math inline">\(\pm\)</span> 73 bps
is usually superposed to the predicted nucleosome to repesent the
nucleosome occupancy.</p>
<p>First, the <em><a href="https://bioconductor.org/packages/3.15/consensusSeekeR" class="external-link">consensusSeekeR</a></em>
package must be loaded.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/adeschen/consensusSeekeR" class="external-link">consensusSeekeR</a></span><span class="op">)</span></span></code></pre></div>
<p>The datasets, which are included in the <em><a href="https://bioconductor.org/packages/3.15/consensusSeekeR" class="external-link">consensusSeekeR</a></em>
package, have to be loaded. Those include results obtained using
syntethic reads distributed following a normal distribution with a
variance of 20 from three different nucleosome positioning software:
PING <span class="citation">(Woo et al. 2013)</span>, NOrMAL <span class="citation">(Polishko et al. 2012)</span> and NucPosSimulator <span class="citation">(Schöpflin et al. 2013)</span>. The genomic ranges have
been obtained by adding <span class="math inline">\(\pm\)</span> 73 bps
to the detected positions.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">### Loading dataset from NOrMAL</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">NOrMAL_nucleosome_positions</span><span class="op">)</span> ; <span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">NOrMAL_nucleosome_ranges</span><span class="op">)</span></span>
<span></span>
<span><span class="co">### Loading dataset from PING</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">PING_nucleosome_positions</span><span class="op">)</span> ; <span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">PING_nucleosome_ranges</span><span class="op">)</span></span>
<span></span>
<span><span class="co">### Loading dataset from NucPosSimulator</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">NucPosSimulator_nucleosome_positions</span><span class="op">)</span> ; <span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">NucPosSimulator_nucleosome_ranges</span><span class="op">)</span></span></code></pre></div>
<p>For the positions and ranges dataset from the same software, the
<code>name</code> field is paired to ensure that each position can be
associated to its range. The metadata field <code>name</code> must be
unique to each feature for all datasets.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">### Each entry in the positions dataset has an equivalent metadata </span></span>
<span><span class="co">### "name" entry in the ranges dataset</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">NOrMAL_nucleosome_positions</span>, n <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="co">## GRanges object with 2 ranges and 1 metadata column:</span></span>
<span><span class="co">##          seqnames    ranges strand |     name</span></span>
<span><span class="co">##             &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;factor&gt;</span></span>
<span><span class="co">##   NOrMAL     chr1     10240      + |      no1</span></span>
<span><span class="co">##   NOrMAL     chr1     10409      + |      no2</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">NOrMAL_nucleosome_ranges</span>, n <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="co">## GRanges object with 2 ranges and 1 metadata column:</span></span>
<span><span class="co">##          seqnames      ranges strand |     name</span></span>
<span><span class="co">##             &lt;Rle&gt;   &lt;IRanges&gt;  &lt;Rle&gt; | &lt;factor&gt;</span></span>
<span><span class="co">##   NOrMAL     chr1 10167-10313      + |      no1</span></span>
<span><span class="co">##   NOrMAL     chr1 10336-10482      + |      no2</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span></code></pre></div>
<p>To be able to identify all entries from the same software, each row
of the dataset has to be assigned a name. All positions and ranges from
the same source must be assigned identical row names. In this exemple,
datasets are going to be identified by the name of their source
software.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">### Assigning software name "NOrMAL" </span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">NOrMAL_nucleosome_positions</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"NOrMAL"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">NOrMAL_nucleosome_positions</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">NOrMAL_nucleosome_ranges</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"NOrMAL"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">NOrMAL_nucleosome_ranges</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">### Assigning experiment name "PING"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">PING_nucleosome_positions</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"PING"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">PING_nucleosome_positions</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">PING_nucleosome_ranges</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"PING"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">PING_nucleosome_ranges</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">### Assigning experiment name "NucPosSimulator"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">NucPosSimulator_nucleosome_positions</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"NucPosSimulator"</span>, </span>
<span>                                <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">NucPosSimulator_nucleosome_positions</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">NucPosSimulator_nucleosome_ranges</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"NucPosSimulator"</span>, </span>
<span>                                <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">NucPosSimulator_nucleosome_ranges</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">### Row names are unique to each software</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">NOrMAL_nucleosome_positions</span>, n <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="co">## GRanges object with 2 ranges and 1 metadata column:</span></span>
<span><span class="co">##          seqnames    ranges strand |     name</span></span>
<span><span class="co">##             &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;factor&gt;</span></span>
<span><span class="co">##   NOrMAL     chr1     10240      + |      no1</span></span>
<span><span class="co">##   NOrMAL     chr1     10409      + |      no2</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">PING_nucleosome_positions</span>, n <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="co">## GRanges object with 2 ranges and 1 metadata column:</span></span>
<span><span class="co">##        seqnames    ranges strand |     name</span></span>
<span><span class="co">##           &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;factor&gt;</span></span>
<span><span class="co">##   PING     chr1     10075      + |       p1</span></span>
<span><span class="co">##   PING     chr1     10241      + |       p2</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">NucPosSimulator_nucleosome_positions</span>, n <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span>
<span><span class="co">## GRanges object with 2 ranges and 1 metadata column:</span></span>
<span><span class="co">##                   seqnames    ranges strand |     name</span></span>
<span><span class="co">##                      &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;factor&gt;</span></span>
<span><span class="co">##   NucPosSimulator     chr1     10075      + |      nu1</span></span>
<span><span class="co">##   NucPosSimulator     chr1     10241      + |      nu2</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span></code></pre></div>
<p>The consensus regions for chromosome 1 only are calculated with a
defaut region size of 50 bases pairs (2 * <code>extendingSize</code>)
The regions are extended to include all nucleosome regions
(<code>expandToFitPeakRegion</code> = <code>TRUE</code> and
<code>shrinkToFitPeakRegion</code> = <code>TRUE</code>). To be retained
as a consensus region, nucleosomes from at least 2 software must be
present in the region (<code>minNbrExp</code> = <code>2</code>). .</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">### Only choromsome 1 is going to be analyzed</span></span>
<span><span class="va">chrList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/Seqinfo-class.html" class="external-link">Seqinfo</a></span><span class="op">(</span><span class="st">"chr1"</span>, <span class="fl">135534747</span>, <span class="cn">NA</span><span class="op">)</span></span>
<span></span>
<span><span class="co">### Find consensus regions with both replicates inside it</span></span>
<span><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/findConsensusPeakRegions.html">findConsensusPeakRegions</a></span><span class="op">(</span></span>
<span>            narrowPeaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">NOrMAL_nucleosome_ranges</span>, <span class="va">PING_nucleosome_ranges</span>,</span>
<span>                                <span class="va">NucPosSimulator_nucleosome_ranges</span><span class="op">)</span>,</span>
<span>            peaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">NOrMAL_nucleosome_positions</span>, <span class="va">PING_nucleosome_positions</span>,</span>
<span>                                <span class="va">NucPosSimulator_nucleosome_positions</span><span class="op">)</span>,</span>
<span>            chrInfo <span class="op">=</span> <span class="va">chrList</span>,</span>
<span>            extendingSize <span class="op">=</span> <span class="fl">25</span>,</span>
<span>            expandToFitPeakRegion <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>            shrinkToFitPeakRegion <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>            minNbrExp <span class="op">=</span> <span class="fl">2</span>,</span>
<span>            nbrThreads <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p>The output of <code>findConsensusPeakRegions</code> function is a
list containing an object <code>call</code> and an object
<code>consensusRanges</code>. The object <code>call</code> contains the
matched call while the object <code>consensusRanges</code> is a
<code>GRanges</code> containing the consensus regions.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">### Print the call </span></span>
<span><span class="va">results</span><span class="op">$</span><span class="va">call</span></span>
<span><span class="co">## findConsensusPeakRegions(narrowPeaks = c(NOrMAL_nucleosome_ranges, </span></span>
<span><span class="co">##     PING_nucleosome_ranges, NucPosSimulator_nucleosome_ranges), </span></span>
<span><span class="co">##     peaks = c(NOrMAL_nucleosome_positions, PING_nucleosome_positions, </span></span>
<span><span class="co">##         NucPosSimulator_nucleosome_positions), chrInfo = chrList, </span></span>
<span><span class="co">##     extendingSize = 25, expandToFitPeakRegion = TRUE, shrinkToFitPeakRegion = TRUE, </span></span>
<span><span class="co">##     minNbrExp = 2, nbrThreads = 1)</span></span>
<span></span>
<span><span class="co">### Print the 3 first consensus regions</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">results</span><span class="op">$</span><span class="va">consensusRanges</span>, n <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span>
<span><span class="co">## GRanges object with 3 ranges and 0 metadata columns:</span></span>
<span><span class="co">##       seqnames      ranges strand</span></span>
<span><span class="co">##          &lt;Rle&gt;   &lt;IRanges&gt;  &lt;Rle&gt;</span></span>
<span><span class="co">##   [1]     chr1 10002-10148      *</span></span>
<span><span class="co">##   [2]     chr1 10167-10314      *</span></span>
<span><span class="co">##   [3]     chr1 10334-10482      *</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span></code></pre></div>
<p>A total of 27 consensus regions have been found. An exemple of the
consensus regions (in dark blue) is shown in Figure
@ref(fig:nucleosomes) using Integrative genomics viewer <span class="citation">(Robinson et al. 2011)</span>:</p>
<div class="figure">
<img src="figures/nucleosomes.jpg" alt=""><p class="caption">Consensus regions. The consensus regions obtained for
peaks called from three software on the same dataset.</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="case-study-chip-seq-data">Case study: ChIP-Seq data<a class="anchor" aria-label="anchor" href="#case-study-chip-seq-data"></a>
</h2>
<p>Next-generation DNA sequencing coupled with chromatin
immunoprecipitation (ChIP-seq) has changed the ability to interrogate
the genomic landscape of histone modifications, transcriptional
cofactors and transcription-factors binding in living cells <span class="citation">(Mundade et al. 2014)</span>. Consortium, such as <a href="https://www.encodeproject.org" class="external-link">ENCODE</a> have developed and are
constantly updating a set of standards and guidelines for ChIP-Seq
experiments <span class="citation">(Landt and Marinov 2012)</span>.</p>
<p>ChIP-seq combines chromatin immunoprecipitation (ChIP) with massively
parallel DNA sequencing. The obtained sequence reads are first mapped to
the reference genome of the organism used in the experiments. Binding
sites are then detected using software specialized in transcript factor
binding sites identification, such as MACS2 <span class="citation">(Zhang et al. 2008)</span> and PeakRanger <span class="citation">(Feng, Grossman, and Stein 2011)</span>. Peaks are
defined as a single base pair position while statistically enriched
regions are defined as genomic ranges.</p>
<div class="section level3">
<h3 id="chip-seq-replicates-from-one-experiment">ChIP-Seq replicates from one experiment<a class="anchor" aria-label="anchor" href="#chip-seq-replicates-from-one-experiment"></a>
</h3>
<p>The Encyclopedia of DNA Elements (ENCODE) Consortium is an
international collaboration of research groups funded by the National
Human Genome Research Institute. The <a href="https://www.encodeproject.org" class="external-link">ENCODE website</a> is a portal
giving access to the data generated by the ENCODE Consortium. The amount
of data gathered is extensive. Moreover, for some experiments, more than
one ChIP-Seq replicate is often available.</p>
<p>The software used to identify transcript factor binding sites
generally generates a peak position and an enriched region for each
binding site. However, it is quite unlikely that the exact peak position
is exactly the same across replicates. Even more, there is not yet a
consensus on how to analyze multiple-replicate ChIP-seq samples <span class="citation">(Yang et al. 2014)</span>.</p>
<p>The <em><a href="https://bioconductor.org/packages/3.15/consensusSeekeR" class="external-link">consensusSeekeR</a></em>
package can be used to identify consensus regions for two or more
replicates ChIP-Seq samples. The consensus regions are being found by
using the peak positions.</p>
<p>The transcription factor binding for the CTCF transcription factor
have been analyzed and 2 replicates are available in BAM files format on
<a href="https://www.encodeproject.org" class="external-link">ENCODE website</a> (DCCs:
ENCFF000MYJ and ENCFF000MYN). The NarrowPeaks were generated using MACS2
<span class="citation">(Zhang et al. 2008)</span> with the default
parameters and the q-value set to 0.05.</p>
<p>To simplify this demo, only part of genome hg19,
chr1:246000000-249250621 and chr10:10000000-12500000, have been retained
in the datasets.</p>
<p>First, the <em><a href="https://bioconductor.org/packages/3.15/consensusSeekeR" class="external-link">consensusSeekeR</a></em>
package must be loaded.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/adeschen/consensusSeekeR" class="external-link">consensusSeekeR</a></span><span class="op">)</span></span></code></pre></div>
<p>The datasets, which are included in the <em><a href="https://bioconductor.org/packages/3.15/consensusSeekeR" class="external-link">consensusSeekeR</a></em>
package, have to be loaded.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">### Loading datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">A549_CTCF_MYN_NarrowPeaks_partial</span><span class="op">)</span> ; <span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">A549_CTCF_MYN_Peaks_partial</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">A549_CTCF_MYJ_NarrowPeaks_partial</span><span class="op">)</span> ; <span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">A549_CTCF_MYJ_Peaks_partial</span><span class="op">)</span></span></code></pre></div>
<p>To be able to identify data from the same source, each row of the
dataset has to be assigned a source name. Beware that
<code>NarrowPeak</code> and <code>Peak</code> datasets from the same
source must be assigned identical names. In this exemple, datasets are
replicates of the same experiment. So, the names “rep01” and “rep02” are
going to be assigned to each dataset.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">### Assigning experiment name "rep01" to the first replicate</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">A549_CTCF_MYJ_NarrowPeaks_partial</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"rep01"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">A549_CTCF_MYJ_NarrowPeaks_partial</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">A549_CTCF_MYJ_Peaks_partial</span><span class="op">)</span>       <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"rep01"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">A549_CTCF_MYJ_Peaks_partial</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">### Assigning experiment name "rep02" to the second replicate</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">A549_CTCF_MYN_NarrowPeaks_partial</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"rep02"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">A549_CTCF_MYN_NarrowPeaks_partial</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">A549_CTCF_MYN_Peaks_partial</span><span class="op">)</span>       <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"rep02"</span>, <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">A549_CTCF_MYN_Peaks_partial</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>The consensus regions for chromosome 10 only are calculated with a
defaut region size of 200 bases pairs (2 * <code>extendingSize</code>)
The regions are extended to include all peaks regions
(<code>expandToFitPeakRegion</code> = <code>TRUE</code> and
<code>shrinkToFitPeakRegion</code> = <code>TRUE</code>). A peak from
both replicates must be present withinin a region for it to be retained
as a consensus region.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">### Only choromsome 10 is going to be analyzed</span></span>
<span><span class="va">chrList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/Seqinfo-class.html" class="external-link">Seqinfo</a></span><span class="op">(</span><span class="st">"chr10"</span>, <span class="fl">135534747</span>, <span class="cn">NA</span><span class="op">)</span></span>
<span></span>
<span><span class="co">### Find consensus regions with both replicates inside it</span></span>
<span><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/findConsensusPeakRegions.html">findConsensusPeakRegions</a></span><span class="op">(</span></span>
<span>            narrowPeaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">A549_CTCF_MYJ_NarrowPeaks_partial</span>, <span class="va">A549_CTCF_MYN_NarrowPeaks_partial</span><span class="op">)</span>,</span>
<span>            peaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">A549_CTCF_MYJ_Peaks_partial</span>, <span class="va">A549_CTCF_MYN_Peaks_partial</span><span class="op">)</span>,</span>
<span>            chrInfo <span class="op">=</span> <span class="va">chrList</span>,</span>
<span>            extendingSize <span class="op">=</span> <span class="fl">100</span>,</span>
<span>            expandToFitPeakRegion <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>            shrinkToFitPeakRegion <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>            minNbrExp <span class="op">=</span> <span class="fl">2</span>,</span>
<span>            nbrThreads <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p>The output of <code>findConsensusPeakRegions</code> function is a
list containing an object <code>call</code> and an object
<code>conesensusRanges</code>. The object <code>call</code> contains the
matched call while the object <code>conesensusRanges</code> is a
<code>GRanges</code> containing the consensus regions.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">### Print the call </span></span>
<span><span class="va">results</span><span class="op">$</span><span class="va">call</span></span>
<span><span class="co">## findConsensusPeakRegions(narrowPeaks = c(A549_CTCF_MYJ_NarrowPeaks_partial, </span></span>
<span><span class="co">##     A549_CTCF_MYN_NarrowPeaks_partial), peaks = c(A549_CTCF_MYJ_Peaks_partial, </span></span>
<span><span class="co">##     A549_CTCF_MYN_Peaks_partial), chrInfo = chrList, extendingSize = 100, </span></span>
<span><span class="co">##     expandToFitPeakRegion = TRUE, shrinkToFitPeakRegion = TRUE, </span></span>
<span><span class="co">##     minNbrExp = 2, nbrThreads = 1)</span></span>
<span></span>
<span><span class="co">### Print the 3 first consensus regions</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">results</span><span class="op">$</span><span class="va">consensusRanges</span>, n <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span>
<span><span class="co">## GRanges object with 3 ranges and 0 metadata columns:</span></span>
<span><span class="co">##       seqnames            ranges strand</span></span>
<span><span class="co">##          &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt;</span></span>
<span><span class="co">##   [1]    chr10 11078504-11078657      *</span></span>
<span><span class="co">##   [2]    chr10 11312608-11312835      *</span></span>
<span><span class="co">##   [3]    chr10 11466554-11466858      *</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span></code></pre></div>
<p>A total of 18 consensus regions have been found. An example of the
consensus regions (in green) is shown in Figure @ref(fig:ctcf), using
Integrative genomics viewer <span class="citation">(Robinson et al.
2011)</span>.</p>
<div class="figure">
<img src="figures/ctcf_consensus.jpg" alt=""><p class="caption">One consensus region for 2 ChIP-Seq replicates.</p>
</div>
</div>
<div class="section level3">
<h3 id="chip-seq-data-from-multiple-experiments">ChIP-Seq data from multiple experiments<a class="anchor" aria-label="anchor" href="#chip-seq-data-from-multiple-experiments"></a>
</h3>
<p>The <em><a href="https://bioconductor.org/packages/3.15/consensusSeekeR" class="external-link">consensusSeekeR</a></em>
package can also be used to identify consensus regions for two or more
ChIP-Seq samples from multiple experiments. The peak positions are the
feature used to identify the consensus regions.</p>
<p>The transcription factor binding for the NR3C1 transcription factor
have been analyzed in more than one experiment. For each experiment,
replicates have been analyzed together using the irreproducible
discovery rate method <span class="citation">(Li et al. 2011)</span>.
Results are available in bed narrowPeak format on <a href="https://www.encodeproject.org" class="external-link">ENCODE website</a> (DCCs:
ENCFF002CFQ, ENCFF002CFR and ENCFF002CFS) <span class="citation">(Dunham
et al. 2012)</span>.</p>
<p>To simplify this demo, only part of genome hg19,
chr2:40000000-50000000 and chr3:10000000-13000000, have been retained in
the datasets.</p>
<p>First, the <em><a href="https://bioconductor.org/packages/3.15/consensusSeekeR" class="external-link">consensusSeekeR</a></em>
package must be loaded.</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/adeschen/consensusSeekeR" class="external-link">consensusSeekeR</a></span><span class="op">)</span></span></code></pre></div>
<p>The datasets, which are included in the <em><a href="https://bioconductor.org/packages/3.15/consensusSeekeR" class="external-link">consensusSeekeR</a></em>
package, have to be loaded.</p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">### Loading datasets</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">A549_NR3C1_CFQ_NarrowPeaks_partial</span><span class="op">)</span> ; <span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">A549_NR3C1_CFQ_Peaks_partial</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">A549_NR3C1_CFR_NarrowPeaks_partial</span><span class="op">)</span> ; <span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">A549_NR3C1_CFR_Peaks_partial</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">A549_NR3C1_CFS_NarrowPeaks_partial</span><span class="op">)</span> ; <span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">A549_NR3C1_CFS_Peaks_partial</span><span class="op">)</span></span></code></pre></div>
<p>To be able to identify data from the same source, each row of the
dataset has to be assigned an experiment name. Beware that NarrowPeak
and Peak datasets from the same source must be assigned identical names.
In this exemple, datasets are coming from different experiments for the
same transcription factor. So, the short name of each experiment
“ENCFF002CFQ”, “ENCFF002CFR” and “ENCFF002CFS” is going to be assigned
to each dataset.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">### Assign experiment name "ENCFF002CFQ" to the first experiment</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">A549_NR3C1_CFQ_NarrowPeaks_partial</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"ENCFF002CFQ"</span>, </span>
<span>                                <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">A549_NR3C1_CFQ_NarrowPeaks_partial</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">A549_NR3C1_CFQ_Peaks_partial</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"ENCFF002CFQ"</span>, </span>
<span>                                <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">A549_NR3C1_CFQ_Peaks_partial</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">### Assign experiment name "ENCFF002CFQ" to the second experiment</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">A549_NR3C1_CFR_NarrowPeaks_partial</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"ENCFF002CFR"</span>, </span>
<span>                                <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">A549_NR3C1_CFR_NarrowPeaks_partial</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">A549_NR3C1_CFR_Peaks_partial</span><span class="op">)</span>       <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"ENCFF002CFR"</span>, </span>
<span>                                <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">A549_NR3C1_CFR_Peaks_partial</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">### Assign experiment name "ENCFF002CFQ" to the third experiment</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">A549_NR3C1_CFS_NarrowPeaks_partial</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"ENCFF002CFS"</span>, </span>
<span>                                <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">A549_NR3C1_CFS_NarrowPeaks_partial</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">A549_NR3C1_CFS_Peaks_partial</span><span class="op">)</span>       <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"ENCFF002CFS"</span>, </span>
<span>                                <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">A549_NR3C1_CFS_Peaks_partial</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>In ENCODE bed narrowPeak format, entries don’t have a specific
metadata field called <code>name</code>. So, to be able to use the
<code>findConsensusPeakRegions</code> function, specific names must
manually be added to each entry.</p>
<div class="sourceCode" id="cb21"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">### Assign specific name to each entry of to first experiment</span></span>
<span><span class="co">### NarrowPeak name must fit Peaks name for same experiment</span></span>
<span><span class="va">A549_NR3C1_CFQ_NarrowPeaks_partial</span><span class="op">$</span><span class="va">name</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"NR3C1_CFQ_region_"</span>, </span>
<span>                                <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">A549_NR3C1_CFQ_NarrowPeaks_partial</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">A549_NR3C1_CFQ_Peaks_partial</span><span class="op">$</span><span class="va">name</span>       <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"NR3C1_CFQ_region_"</span>,</span>
<span>                                <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">A549_NR3C1_CFQ_NarrowPeaks_partial</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">### Assign specific name to each entry of to second experiment</span></span>
<span><span class="co">### NarrowPeak name must fit Peaks name for same experiment</span></span>
<span><span class="va">A549_NR3C1_CFR_NarrowPeaks_partial</span><span class="op">$</span><span class="va">name</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"NR3C1_CFR_region_"</span>, </span>
<span>                                <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">A549_NR3C1_CFR_NarrowPeaks_partial</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">A549_NR3C1_CFR_Peaks_partial</span><span class="op">$</span><span class="va">name</span>       <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"NR3C1_CFR_region_"</span>, </span>
<span>                                <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">A549_NR3C1_CFR_Peaks_partial</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">### Assign specific name to each entry of to third experiment</span></span>
<span><span class="co">### NarrowPeak name must fit Peaks name for same experiment</span></span>
<span><span class="va">A549_NR3C1_CFS_NarrowPeaks_partial</span><span class="op">$</span><span class="va">name</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"NR3C1_CFS_region_"</span>, </span>
<span>                                <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">A549_NR3C1_CFS_NarrowPeaks_partial</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="va">A549_NR3C1_CFS_Peaks_partial</span><span class="op">$</span><span class="va">name</span>       <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"NR3C1_CFS_region_"</span>, </span>
<span>                                <span class="fl">1</span><span class="op">:</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">A549_NR3C1_CFS_Peaks_partial</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>The consensus regions for chromosome 2 only are calculated with a
defaut region size of 400 bases pairs (2 * <code>extendingSize</code>)
The regions are not extended to include all peaks regions but are
shrinked when exceeding peaks regions
(<code>expandToFitPeakRegion</code> = <code>FALSE</code> and
<code>shrinkToFitPeakRegion</code> = <code>TRUE</code>). A peak from 2
out of 3 experiments must be present in a region for it to be retained
as a consensus region.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">### Only choromsome 2 is going to be analyzed</span></span>
<span><span class="va">chrList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/Seqinfo-class.html" class="external-link">Seqinfo</a></span><span class="op">(</span><span class="st">"chr2"</span>, <span class="fl">243199373</span>, <span class="cn">NA</span><span class="op">)</span></span>
<span></span>
<span><span class="co">### Find consensus regions with both replicates inside it</span></span>
<span><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/findConsensusPeakRegions.html">findConsensusPeakRegions</a></span><span class="op">(</span></span>
<span>            narrowPeaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">A549_NR3C1_CFQ_NarrowPeaks_partial</span>, </span>
<span>                        <span class="va">A549_NR3C1_CFR_NarrowPeaks_partial</span>,</span>
<span>                        <span class="va">A549_NR3C1_CFS_NarrowPeaks_partial</span><span class="op">)</span>,</span>
<span>            peaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">A549_NR3C1_CFQ_Peaks_partial</span>, </span>
<span>                        <span class="va">A549_NR3C1_CFR_Peaks_partial</span>,</span>
<span>                        <span class="va">A549_NR3C1_CFS_Peaks_partial</span><span class="op">)</span>,</span>
<span>            chrInfo <span class="op">=</span> <span class="va">chrList</span>,</span>
<span>            extendingSize <span class="op">=</span> <span class="fl">200</span>,</span>
<span>            expandToFitPeakRegion <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>            shrinkToFitPeakRegion <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>            minNbrExp <span class="op">=</span> <span class="fl">2</span>,</span>
<span>            nbrThreads <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p>The output of <code>findConsensusPeakRegions</code> function is a
list containing an object <code>call</code> and an object
<code>consensusRanges</code>. The object <code>call</code> contains the
matched call while the object <code>consensusRanges</code> is a
<code>GRanges</code> containing the consensus regions.</p>
<div class="sourceCode" id="cb23"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">### Print the call </span></span>
<span><span class="va">results</span><span class="op">$</span><span class="va">call</span></span>
<span><span class="co">## findConsensusPeakRegions(narrowPeaks = c(A549_NR3C1_CFQ_NarrowPeaks_partial, </span></span>
<span><span class="co">##     A549_NR3C1_CFR_NarrowPeaks_partial, A549_NR3C1_CFS_NarrowPeaks_partial), </span></span>
<span><span class="co">##     peaks = c(A549_NR3C1_CFQ_Peaks_partial, A549_NR3C1_CFR_Peaks_partial, </span></span>
<span><span class="co">##         A549_NR3C1_CFS_Peaks_partial), chrInfo = chrList, extendingSize = 200, </span></span>
<span><span class="co">##     expandToFitPeakRegion = FALSE, shrinkToFitPeakRegion = TRUE, </span></span>
<span><span class="co">##     minNbrExp = 2, nbrThreads = 1)</span></span>
<span></span>
<span><span class="co">### Print the first 3 consensus regions</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">results</span><span class="op">$</span><span class="va">consensusRanges</span>, n <span class="op">=</span> <span class="fl">3</span><span class="op">)</span></span>
<span><span class="co">## GRanges object with 3 ranges and 0 metadata columns:</span></span>
<span><span class="co">##       seqnames            ranges strand</span></span>
<span><span class="co">##          &lt;Rle&gt;         &lt;IRanges&gt;  &lt;Rle&gt;</span></span>
<span><span class="co">##   [1]     chr2 42054830-42055168      *</span></span>
<span><span class="co">##   [2]     chr2 42075261-42075544      *</span></span>
<span><span class="co">##   [3]     chr2 42153872-42154111      *</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span></code></pre></div>
<p>A total of 99 consensus regions have been found. A example of the
consensus regions (in green) is shown in Figure @ref(fig:NR3C1) using
Integrative genomics viewer <span class="citation">(Robinson et al.
2011)</span>.</p>
<div class="figure">
<img src="figures/NR3C1_consensus.jpg" alt=""><p class="caption">Consensus regions. The consensus regions obtained for
peaks called for 3 NR2C1 ChIP-Seq experiments.</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="parameters">Parameters<a class="anchor" aria-label="anchor" href="#parameters"></a>
</h2>
<div class="section level3">
<h3 id="effect-of-the-shrinktofitpeakregion-parameter">Effect of the shrinkToFitPeakRegion parameter<a class="anchor" aria-label="anchor" href="#effect-of-the-shrinktofitpeakregion-parameter"></a>
</h3>
<p>The <code>shrinkToFitPeakRegion</code> allows the resizing of the
consensus region to fit the minimum regions of the included features
when those values are included inside the initial consensus region. When
the <code>extendingSize</code> parameter is large, the effect can be
quite visible on the final consensus regions. For exemple, Figure
@ref(fig:shrink) shows the same region, from the NR3C1 example with
<code>extendingSize</code> of 200, when the
<code>shrinkToFitPeakRegion</code> is set to <code>TRUE</code> (green
color) and to <code>FALSE</code> (orange color) using Integrative
genomics viewer <span class="citation">(Robinson et al.
2011)</span>.</p>
<div class="figure">
<img src="figures/shrink.jpg" alt=""><p class="caption">Effect of the shrinkToFitPeakRegion parameter.</p>
</div>
</div>
<div class="section level3">
<h3 id="effect-of-the-expandtofitpeakregion-parameter">Effect of the expandToFitPeakRegion parameter<a class="anchor" aria-label="anchor" href="#effect-of-the-expandtofitpeakregion-parameter"></a>
</h3>
<p>The <code>expandToFitPeakRegion</code> allows the resizing of the
consensus region to fit the maximum of the included features when those
values are outside the initial consensus region. When the
<code>extendingSize</code> parameter is small, the effect can be quite
visible on the final consensus regions. For exemple, the following
figure shows the same region, from the CTCF example with
<code>extendingSize</code> of 100, when the
<code>shrinkToFitPeakRegion</code> is set to <code>FALSE</code> (orange
color) and to <code>TRUE</code> (green color).</p>
<div class="figure">
<img src="figures/extendNew.jpg" alt=""><p class="caption">Effect of the expandToFitPeakRegion parameter.</p>
</div>
</div>
<div class="section level3">
<h3 id="effect-of-the-extendingsize-parameter">Effect of the extendingSize parameter<a class="anchor" aria-label="anchor" href="#effect-of-the-extendingsize-parameter"></a>
</h3>
<p>The value of the <code>extendingSize</code> parameter can affect the
final number of consensus regions. While small
<code>extendingSize</code> value can miss some regions, large
<code>extendingSize</code> value can gather consensus regions. Testing a
range of <code>extendingSize</code> parameters can be an option worth
considering.</p>
<p>As an example, the number of consensus regions obtained with
different values of <code>extendingSize</code> is calculated.</p>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">### Set different values for the extendingSize parameter</span></span>
<span><span class="va">size</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>, <span class="fl">10</span>, <span class="fl">50</span>, <span class="fl">100</span>, <span class="fl">300</span>, <span class="fl">500</span>, <span class="fl">750</span>, <span class="fl">1000</span><span class="op">)</span></span>
<span></span>
<span><span class="co">### Only chrompsome 10 is going to be analyzed</span></span>
<span><span class="va">chrList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/Seqinfo-class.html" class="external-link">Seqinfo</a></span><span class="op">(</span><span class="st">"chr10"</span>, <span class="fl">135534747</span>, <span class="cn">NA</span><span class="op">)</span></span>
<span></span>
<span><span class="co">### Find consensus regions using all the size values</span></span>
<span><span class="va">resultsBySize</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/lapply.html" class="external-link">lapply</a></span><span class="op">(</span><span class="va">size</span>, FUN <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">size</span><span class="op">)</span> <span class="fu"><a href="../reference/findConsensusPeakRegions.html">findConsensusPeakRegions</a></span><span class="op">(</span></span>
<span>                    narrowPeaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">A549_CTCF_MYJ_NarrowPeaks_partial</span>, </span>
<span>                                    <span class="va">A549_CTCF_MYN_NarrowPeaks_partial</span><span class="op">)</span>,</span>
<span>                    peaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">A549_CTCF_MYJ_Peaks_partial</span>, </span>
<span>                                    <span class="va">A549_CTCF_MYN_Peaks_partial</span><span class="op">)</span>,</span>
<span>                    chrInfo <span class="op">=</span> <span class="va">chrList</span>,</span>
<span>                    extendingSize <span class="op">=</span> <span class="va">size</span>,</span>
<span>                    expandToFitPeakRegion <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                    shrinkToFitPeakRegion <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                    minNbrExp <span class="op">=</span> <span class="fl">2</span>,</span>
<span>                    nbrThreads <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">### Extract the number of consensus regions obtained for each extendingSize</span></span>
<span><span class="va">nbrRegions</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/BiocGenerics/man/mapply.html" class="external-link">mapply</a></span><span class="op">(</span><span class="va">resultsBySize</span>, </span>
<span>                FUN <span class="op">=</span> <span class="kw">function</span><span class="op">(</span><span class="va">x</span><span class="op">)</span> <span class="kw"><a href="https://rdrr.io/r/base/function.html" class="external-link">return</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">x</span><span class="op">$</span><span class="va">consensusRanges</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p>A graph can be used to visualize the variation of the number of
consensus regions in function of the <code>extendingSize</code>
parameter (see Figure @ref(fig:sizeEffectG)).</p>
<div class="sourceCode" id="cb25"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html" class="external-link">data.frame</a></span><span class="op">(</span>extendingSize <span class="op">=</span> <span class="va">size</span>, nbrRegions <span class="op">=</span> <span class="va">nbrRegions</span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">data</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">extendingSize</span>, <span class="va">nbrRegions</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_x_log10</a></span><span class="op">(</span><span class="st">"Extending size"</span><span class="op">)</span> <span class="op">+</span> </span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_smooth.html" class="external-link">stat_smooth</a></span><span class="op">(</span>se <span class="op">=</span> <span class="cn">FALSE</span>, method <span class="op">=</span> <span class="st">"loess"</span>, size<span class="op">=</span><span class="fl">1.4</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ylab</a></span><span class="op">(</span><span class="st">"Number of consensus regions"</span><span class="op">)</span> <span class="op">+</span></span>
<span>        <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">ggtitle</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"Number of consensus regions in function of the extendingSize"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<div class="figure">
<img src="figures/sizeEffect.jpg" alt=""><p class="caption">Effect of the extendingSize parameter.</p>
</div>
</div>
</div>
<div class="section level2">
<h2 id="parallelizing-consensusseeker">Parallelizing consensusSeekeR<a class="anchor" aria-label="anchor" href="#parallelizing-consensusseeker"></a>
</h2>
<p>Due to the size of the analyzed genomes, the
<code>findConsensusPeakRegions</code> function can take a while to
process. However, a job can be separated by chromosome and run in
parallel. This takes advantage of multiple processors and reduce the
total execution time. The number of threads used can be set with the
<code>nbrThreads</code> parameter in the
<code>findConsensusPeakRegions</code> function.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">### Load data</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">A549_FOSL2_01_NarrowPeaks_partial</span><span class="op">)</span> ; <span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">A549_FOSL2_01_Peaks_partial</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">A549_FOXA1_01_NarrowPeaks_partial</span><span class="op">)</span> ; <span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="va">A549_FOXA1_01_Peaks_partial</span><span class="op">)</span></span>
<span></span>
<span><span class="co">### Assigning name "FOSL2" </span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">A549_FOSL2_01_NarrowPeaks_partial</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"FOSL2"</span>, </span>
<span>                                    <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">A549_FOSL2_01_NarrowPeaks_partial</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">A549_FOSL2_01_Peaks_partial</span><span class="op">)</span>       <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"FOSL2"</span>, </span>
<span>                                    <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">A549_FOSL2_01_Peaks_partial</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">### Assigning name "FOXA1"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">A549_FOXA1_01_NarrowPeaks_partial</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"FOXA1"</span>, </span>
<span>                                    <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">A549_FOXA1_01_NarrowPeaks_partial</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">A549_FOXA1_01_Peaks_partial</span><span class="op">)</span>       <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rep.html" class="external-link">rep</a></span><span class="op">(</span><span class="st">"FOXA1"</span>, </span>
<span>                                    <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="va">A549_FOXA1_01_Peaks_partial</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">### Two chromosomes to analyse</span></span>
<span><span class="va">chrList</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/GenomeInfoDb/man/Seqinfo-class.html" class="external-link">Seqinfo</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"chr"</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">10</span><span class="op">)</span><span class="op">)</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">249250621</span>, <span class="fl">135534747</span><span class="op">)</span>, <span class="cn">NA</span><span class="op">)</span></span>
<span></span>
<span><span class="co">### Find consensus regions using 2 threads</span></span>
<span><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/findConsensusPeakRegions.html">findConsensusPeakRegions</a></span><span class="op">(</span></span>
<span>            narrowPeaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">A549_FOSL2_01_NarrowPeaks_partial</span>, </span>
<span>                            <span class="va">A549_FOXA1_01_Peaks_partial</span><span class="op">)</span>,</span>
<span>            peaks <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="va">A549_FOSL2_01_Peaks_partial</span>, </span>
<span>                            <span class="va">A549_FOXA1_01_NarrowPeaks_partial</span><span class="op">)</span>,</span>
<span>            chrInfo <span class="op">=</span> <span class="va">chrList</span>, extendingSize <span class="op">=</span> <span class="fl">200</span>, minNbrExp <span class="op">=</span> <span class="fl">2</span>, </span>
<span>            expandToFitPeakRegion <span class="op">=</span> <span class="cn">FALSE</span>, shrinkToFitPeakRegion <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>            nbrThreads <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="acknowledgment">Acknowledgment<a class="anchor" aria-label="anchor" href="#acknowledgment"></a>
</h2>
<p>We thank Imène Boudaoud for her advice on the vignette content.</p>
</div>
<div class="section level2">
<h2 id="session-info">Session info<a class="anchor" aria-label="anchor" href="#session-info"></a>
</h2>
<p>Here is the output of <code><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo()</a></code> on the system on
which this document was compiled:</p>
<pre><code><span><span class="co">## R version 4.2.1 (2022-06-23)</span></span>
<span><span class="co">## Platform: x86_64-pc-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Ubuntu 20.04.4 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3</span></span>
<span><span class="co">## LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/liblapack.so.3</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] stats4    stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">##  [1] rtracklayer_1.56.1     consensusSeekeR_1.31.2 BiocParallel_1.30.4   </span></span>
<span><span class="co">##  [4] GenomicRanges_1.48.0   GenomeInfoDb_1.32.4    IRanges_2.30.1        </span></span>
<span><span class="co">##  [7] S4Vectors_0.34.0       BiocGenerics_0.42.0    knitr_1.45            </span></span>
<span><span class="co">## [10] BiocStyle_2.24.0      </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##  [1] SummarizedExperiment_1.26.1 xfun_0.42                  </span></span>
<span><span class="co">##  [3] bslib_0.6.1                 purrr_1.0.2                </span></span>
<span><span class="co">##  [5] lattice_0.22-5              vctrs_0.6.5                </span></span>
<span><span class="co">##  [7] htmltools_0.5.7             yaml_2.3.8                 </span></span>
<span><span class="co">##  [9] XML_3.99-0.16.1             rlang_1.1.3                </span></span>
<span><span class="co">## [11] pkgdown_2.0.7               jquerylib_0.1.4            </span></span>
<span><span class="co">## [13] glue_1.7.0                  matrixStats_1.2.0          </span></span>
<span><span class="co">## [15] GenomeInfoDbData_1.2.8      lifecycle_1.0.4            </span></span>
<span><span class="co">## [17] MatrixGenerics_1.8.1        stringr_1.5.1              </span></span>
<span><span class="co">## [19] zlibbioc_1.42.0             Biostrings_2.64.1          </span></span>
<span><span class="co">## [21] ragg_1.2.7                  codetools_0.2-19           </span></span>
<span><span class="co">## [23] memoise_2.0.1               evaluate_0.23              </span></span>
<span><span class="co">## [25] restfulr_0.0.15             Biobase_2.56.0             </span></span>
<span><span class="co">## [27] fastmap_1.1.1               parallel_4.2.1             </span></span>
<span><span class="co">## [29] highr_0.10                  BiocManager_1.30.22        </span></span>
<span><span class="co">## [31] DelayedArray_0.22.0         cachem_1.0.8               </span></span>
<span><span class="co">## [33] desc_1.4.3                  jsonlite_1.8.8             </span></span>
<span><span class="co">## [35] XVector_0.36.0              systemfonts_1.0.5          </span></span>
<span><span class="co">## [37] fs_1.6.3                    Rsamtools_2.12.0           </span></span>
<span><span class="co">## [39] textshaping_0.3.7           rjson_0.2.21               </span></span>
<span><span class="co">## [41] digest_0.6.34               stringi_1.8.3              </span></span>
<span><span class="co">## [43] bookdown_0.37               grid_4.2.1                 </span></span>
<span><span class="co">## [45] BiocIO_1.6.0                cli_3.6.2                  </span></span>
<span><span class="co">## [47] tools_4.2.1                 bitops_1.0-7               </span></span>
<span><span class="co">## [49] magrittr_2.0.3              sass_0.4.8                 </span></span>
<span><span class="co">## [51] RCurl_1.98-1.14             crayon_1.5.2               </span></span>
<span><span class="co">## [53] Matrix_1.6-5                rmarkdown_2.25             </span></span>
<span><span class="co">## [55] R6_2.5.1                    GenomicAlignments_1.32.1   </span></span>
<span><span class="co">## [57] compiler_4.2.1</span></span></code></pre>
</div>
<div class="section level2">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent">
<div id="ref-Dunham2012a" class="csl-entry">
Dunham, Ian, Anshul Kundaje, Shelley F. Aldred, Patrick J. Collins,
Carrie A. Davis, Francis Doyle, Charles B. Epstein, et al. 2012.
<span>“<span class="nocase">An integrated encyclopedia of DNA elements
in the human genome</span>.”</span> <em>Nature</em> 489 (7414): 57–74.
<a href="https://doi.org/10.1038/nature11247" class="external-link">https://doi.org/10.1038/nature11247</a>.
</div>
<div id="ref-Feng2011" class="csl-entry">
Feng, Xin, Robert Grossman, and Lincoln Stein. 2011. <span>“<span class="nocase">PeakRanger: a cloud-enabled peak caller for ChIP-seq
data.</span>”</span> <em>BMC Bioinformatics</em> 12 (1): 139. <a href="https://doi.org/10.1186/1471-2105-12-139" class="external-link">https://doi.org/10.1186/1471-2105-12-139</a>.
</div>
<div id="ref-Jiang2015" class="csl-entry">
Jiang, Jiming, Tao Zhang, and Wenli Zhang. 2015. <span>“<span class="nocase">Genome-wide nucleosome occupancy and positioning and
their impact on gene expression and evolution in plants</span>.”</span>
<em>Plant Physiology</em>, pp.00125.2015. <a href="https://doi.org/10.1104/pp.15.00125" class="external-link">https://doi.org/10.1104/pp.15.00125</a>.
</div>
<div id="ref-Kornberg1999" class="csl-entry">
Kornberg, Roger D., and Yahli Lorch. 1999. <span>“<span class="nocase">Twenty-five years of the nucleosome, fundamental particle
of the eukaryote chromosome</span>.”</span> <em>Cell</em> 98 (3):
285–94. <a href="https://doi.org/10.1016/S0092-8674(00)81958-3" class="external-link">https://doi.org/10.1016/S0092-8674(00)81958-3</a>.
</div>
<div id="ref-Landt2012" class="csl-entry">
Landt, Sg, and Gk Marinov. 2012. <span>“<span class="nocase">ChIP-seq
guidelines and practices of the ENCODE and modENCODE
consortia</span>.”</span> <em>Genome Res.</em>, no. Park 2009: 1813–31.
<a href="https://doi.org/10.1101/gr.136184.111." class="external-link">https://doi.org/10.1101/gr.136184.111.</a>
</div>
<div id="ref-Lawrence2013" class="csl-entry">
Lawrence, Michael, Wolfgang Huber, Hervé Pagès, Patrick Aboyoun, Marc
Carlson, Robert Gentleman, Martin T. Morgan, and Vincent J. Carey. 2013.
<span>“<span class="nocase">Software for Computing and Annotating
Genomic Ranges</span>.”</span> <em>PLoS Computational Biology</em> 9
(8). <a href="https://doi.org/10.1371/journal.pcbi.1003118" class="external-link">https://doi.org/10.1371/journal.pcbi.1003118</a>.
</div>
<div id="ref-Li2011" class="csl-entry">
Li, Qunhua, James B. Brown, Haiyan Huang, and Peter J. Bickel. 2011.
<span>“<span class="nocase">Measuring reproducibility of high-throughput
experiments</span>.”</span> <em>Annals of Applied Statistics</em> 5 (3):
1752–79. <a href="https://doi.org/10.1214/11-AOAS466" class="external-link">https://doi.org/10.1214/11-AOAS466</a>.
</div>
<div id="ref-Mundade2014" class="csl-entry">
Mundade, Rasika, Hatice Gulcin Ozer, Han Wei, Lakshmi Prabhu, and Tao
Lu. 2014. <span>“<span class="nocase">Role of ChIP-seq in the discovery
of transcription factor binding sites, differential gene regulation
mechanism, epigenetic marks and beyond</span>.”</span> <em>Cell
Cycle</em> 13 (18): 2847–52. <a href="https://doi.org/10.4161/15384101.2014.949201" class="external-link">https://doi.org/10.4161/15384101.2014.949201</a>.
</div>
<div id="ref-Polishko2012" class="csl-entry">
Polishko, Anton, Nadia Ponts, Karine G. Le Roch, and Stefano Lonardi.
2012. <span>“<span class="nocase">NOrMAL: Accurate nucleosome
positioning using a modified Gaussian mixture model</span>.”</span>
<em>Bioinformatics</em> 28 (12): 242–49. <a href="https://doi.org/10.1093/bioinformatics/bts206" class="external-link">https://doi.org/10.1093/bioinformatics/bts206</a>.
</div>
<div id="ref-Interests2011" class="csl-entry">
Robinson, James T., Thorvaldsdóttir Helga, Wendy Winckler, Mitchell
Guttman, Eric S. Lander, Gad Getz, and Jill P. Mesirov. 2011.
<span>“<span class="nocase">Integrative genomics viewer</span>.”</span>
<em>Nature Biotechnology</em> 29 (1): 24–26. <a href="https://doi.org/10.1038/nbt0111-24" class="external-link">https://doi.org/10.1038/nbt0111-24</a>.
</div>
<div id="ref-Schopflin2013" class="csl-entry">
Schöpflin, Robert, Vladimir B. Teif, Oliver Müller, Christin Weinberg,
Karsten Rippe, and Gero Wedemann. 2013. <span>“<span class="nocase">Modeling nucleosome position distributions from
experimental nucleosome positioning maps</span>.”</span>
<em>Bioinformatics</em> 29 (19): 2380–86. <a href="https://doi.org/10.1093/bioinformatics/btt404" class="external-link">https://doi.org/10.1093/bioinformatics/btt404</a>.
</div>
<div id="ref-Woo2013" class="csl-entry">
Woo, Sangsoon, Xuekui Zhang, Renan Sauteraud, François Robert, and
Raphael Gottardo. 2013. <span>“<span class="nocase">PING 2.0: An
R/Bioconductor package for nucleosome positioning using next-generation
sequencing data</span>.”</span> <em>Bioinformatics</em> 29 (16):
2049–50. <a href="https://doi.org/10.1093/bioinformatics/btt348" class="external-link">https://doi.org/10.1093/bioinformatics/btt348</a>.
</div>
<div id="ref-Yang2014" class="csl-entry">
Yang, Yajie, Justin Fear, Jianhong Hu, Irina Haecker, Lei Zhou, Rolf
Renne, David Bloom, and Lauren M McIntyre. 2014. <span>“<span class="nocase">Leveraging biological replicates to improve analysis in
ChIP-seq experiments.</span>”</span> <em>Computational and Structural
Biotechnology Journal</em> 9: e201401002. <a href="https://doi.org/10.5936/csbj.201401002" class="external-link">https://doi.org/10.5936/csbj.201401002</a>.
</div>
<div id="ref-Zhang2008" class="csl-entry">
Zhang, Yong, Tao Liu, Clifford a Meyer, Jérôme Eeckhoute, David S
Johnson, Bradley E Bernstein, Chad Nusbaum, et al. 2008. <span>“<span class="nocase">Model-based analysis of ChIP-Seq (MACS).</span>”</span>
<em>Genome Biology</em> 9 (9): R137. <a href="https://doi.org/10.1186/gb-2008-9-9-r137" class="external-link">https://doi.org/10.1186/gb-2008-9-9-r137</a>.
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by Astrid Deschênes, Pascal Belleau, Arnaud Droit.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
